go/connection.go

```diff
@@ -47,6 +51,67 @@ type Connection struct {
 	numInput  int
 }
 
+// buildExecutionParams converts Go data types into strings for query arguments in parameterized queries.
+func (c *Connection) buildExecutionParams(args []driver.Value) ([]*string, error) {
+	executionParams := []*string{}
+	for _, arg := range args {
+		if arg == nil {
+			val := "NULL"
+			executionParams = append(executionParams, aws.String(val))
+			continue
+		}
+		// type switches of arg to handle different query parameter types
+		val := ""
+		switch v := arg.(type) {
+		case int64:
+			val = strconv.FormatInt(v, 10)
+		case uint64:
+			val = strconv.FormatUint(v, 10)
+		case float64:
+			val = strconv.FormatFloat(v, 'g', -1, 64)
+		case bool:
+			if v {
+				val = "1"
+			} else {
+				val = "0"
+			}
+		case time.Time:
+			// Note: time.Time objects are transformed into strings for a STRING/CHAR/VARCHAR type column.
+			// To maintain compatibility with the current interpolateParams() behavior, this function produces a string
+			// up to microsecond granularity, which Athena does not support in TIMESTAMP columns (up to milliseconds).
+			// For DATE/TIME/TIMESTAMP, it is better to pass in string arguments with a typecast. Refer to the string
+			// case below.
+			val = "'0000-00-00'"
+			if !v.IsZero() {
+				v := v.In(time.UTC)
+				v = v.Add(time.Nanosecond * 500) // To round under microsecond
+				dateFormat := timestampFormatDriverMicro
+				if v.Nanosecond()/1000 == 0 {
+					// Omit microseconds if that part is zero
+					dateFormat = time.DateTime
+				}
+				val = fmt.Sprintf("'%s'", v.Format(dateFormat))
+			}
+		case []byte:
+			// Note: Different from interpolateParams() behavior.
+			// Like the string case below, enclosing in single quotes would prevent typecasting or function calls in
+			// execution parameters. Prior to passing in query arguments, Format* functions in utils.go can be used.
+			val = string(v)
```

Comment:
Do we have to do any special character handling here?

Reply:
Ah, like the `string` case, I opted to provide a `FormatBytes()` string for handling special characters before passing in query parameters in. 

Hm... I guess alternatively, we could have this function handle special characters, but provide functions for users to properly enclose their byte slices / strings. Trying to think of a case where handling special characters conflicts with typecasting or function calls, but none come to mind at the moment.